features/nfc/nfc/NFCEEPROMDriver.h

```diff
@@ -0,0 +1,188 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MBED_NFC_EEPROM_DRIVER_H
+#define MBED_NFC_EEPROM_DRIVER_H
+
+#include <stdint.h>
+
+#include "NFCDefinitions.h"
+#include "NFCTarget.h"
+#include "NFCEEPROMDriver.h"
+
+namespace mbed {
+namespace nfc {
+
+    /**
+     * @addtogroup nfc
+     * @{
+     */
+    
+    /**
+     * The abstraction for a NFC EEPROM driver.
+     * Implementers need to derive from this class and implement its methods.
+     */ 
+    class NFCEEPROMDriver  {
+    public:
+        /**
+         * Construct a NFCEEPROM driver instance.
+         */
+        NFCEEPROMDriver();
+        
+        /**
+         * The NFCEEPROMDriver delegate.
+         * Methods in this class are called by the driver on completion of long-running operations.
+         */
+        struct Delegate {
+            /**
+             * Completion of session start operation.
+             * 
+             * @param[in] success whether this operation succeeded
+             */  
+            virtual void has_started_session(bool success) = 0;
+
+            /**
+             * Completion of session end operation.
+             * 
+             * @param[in] success whether this operation succeeded
+             */  
+            virtual void has_ended_session(bool success) = 0;
+
+            /**
+             * Completion of read operation.
+             * 
+             * @param[in] success whether this operation succeeded
+             */  
+            virtual void has_read_bytes(bool success) = 0;
+
+            /**
+             * Completion of write operation.
+             * 
+             * @param[in] success whether this operation succeeded
+             */  
+            virtual void has_written_bytes(bool success) = 0;
+
+            /**
+             * Completion of size setting operation.
+             * 
+             * @param[in] success whether this operation succeeded
+             */  
+            virtual void has_set_size(bool success) = 0;
+
+            /**
+             * Completion of size retrieval operation.
+             * 
+             * @param[in] success whether this operation succeeded
+             * @param[out] the current addressable memory size
+             */  
+            virtual void has_gotten_size(bool success, size_t size) = 0;
+
+            /**
+             * Completion of erasing operation.
+             * 
+             * @param[in] success whether this operation succeeded
+             */  
+            virtual void has_erased_bytes(bool success) = 0;
```

Comment:
"has erased bytes" just reads weirdly, bytes_erased is perfectly fine and if you ask Vincent he'll tell you it should be:
on_bytes_erased()

My suggestion:
``` C++
       virtual void on_session_start(bool success) { };
       virtual void on_session_end(bool success) { };
       virtual void on_bytes_read(bool success) { };
       virtual void on_bytes_written(bool success) { };
       virtual void on_bytes_erased(bool success) { };
       virtual void on_size_read(bool success, size_t size) { };
       virtual void on_size_set(bool success) { };
```